src/components/Breadcrumbs/index.tsx

```diff
@@ -0,0 +1,55 @@
+import { ComponentProps, ReactNode } from "react";
+import clsx from "clsx";
+import { LegacyChevronRight1pxIcon } from "@deriv/quill-icons";
+import { Text } from "../Text";
+import "./Breadcrumbs.scss";
+
+type TItem = {
+    value: string | number;
+    text: string | JSX.Element;
+};
```

Comment:
just a suggestion. 

```suggestion
type TItem = {
    text: string | JSX.Element;
};
```

and then have a function for generating the key:

```
const getKey = (content: React.ReactNode): string => {
  const extractedText = reactNodeToString(content);
  return extractedText || `item-${Math.random().toString(36).substr(2, 9)}`;
};

const reactNodeToString = function (reactNode: React.ReactNode): string {
        let string = "";
        if (typeof reactNode === "string") {
            string = reactNode;
        } else if (typeof reactNode === "number") {
            string = reactNode.toString();
        } else if (reactNode instanceof Array) {
            reactNode.forEach(function (child) {
                string += reactNodeToString(child);
            });
        } else if (isValidElement(reactNode)) {
            string += reactNodeToString(reactNode.props.children);
        }
        return string;
    };
```
and then use it as the key:
```
<li key={getKey(item.content)}... />
```

Reply:
@shayan-deriv wouldn't generating keys using Math.random() cause unpredictable behaviour and lead to rerendering issues? If the concern is having the key to be unique for the specific component maybe it can be something like this instead:
```
key={`deriv-breadcrumb-${item.value}`}
```

Stackoverflow reference: https://stackoverflow.com/questions/29808636/when-giving-unique-keys-to-components-is-it-okay-to-use-math-random-for-gener

Reply:
hey @shayan-deriv, not sure that generating keys on the fly is a good idea, actually this is an anti-pattern which can lead to performance issues.

Reply:
@heorhi-deriv @lubega-deriv you're absolutely right. I didn't think about that point. thanks guys